### ISSUE TITLE
Add NUnit console to Scoop

### ISSUE THREAD

**Issue by unickq** [nunit/nunit-console#509]
https://github.com/lukesampson/scoop

Should be very usefull 👍

**Comment by mikkelbu** [comment]
Hi @unickq 
I had not heard about scoop before, but it seems that one can just create a json with the correct information in https://github.com/lukesampson/scoop-extras and then nunit3-console would be available in the extras bucket. So at least for me it seems that this request should be raised in the scoop-extras project.

**Comment by unickq** [comment]
@mikkelbu

Yep. It’s just a JSON - the thing is a permission from NUnit authors for someone who wants to publish it there. 
Or they can maintain it as a part of distribution (like Nuget)

**Comment by CharliePoole** [comment]
It's similar to chocolatey, I would imagine, where anyone can create packages that use the published installer but it takes a special review to include the binaries as we do.

Does the json just reference a download URL?

**Comment by unickq** [comment]
Yep. Made one like this:

```json
{
    "homepage": "https://nunit.org/",
    "description": "NUnit Console runner and test engine - NUnit, the most popular and widely used unit testing framework for .NET",
    "version": "3.9",
    "license": "MIT",
    "url": "https://github.com/nunit/nunit-console/releases/download/v3.9/NUnit.Console-3.9.0.zip",
    "hash": "EE11B97BD075414EC196ED6EAE8E526AA78BB59251D9BDABE513D07E3B2B7BEA",
    "bin": "nunit3-console.exe",
    "suggest": {
        "NuGet": [
            "nuget"
        ]
    },
    "checkver": {
        "github": "https://github.com/nunit/nunit-console"
    },
    "autoupdate": {
	"url": "https://github.com/nunit/nunit-console/releases/download/v$version/NUnit.Console-$version.0.zip"
    }
}
```

**Comment by unickq** [comment]
So let's add it there if you don't mind 😃

**Comment by rprouse** [comment]
You are welcome to add a pull request for this. I would suggest that your `version` variable be 3.9.0 though so that your auto-update url doesn't have the `.0` in `NUnit.Console-$version.0.zip`. We occasionally release hotfix versions like `3.9.1` and I think that would break your auto update.

**Comment by unickq** [comment]
Merged https://github.com/lukesampson/scoop/pull/2871 - works amazingly, you can try
`scoop install nunit`

@rprouse 
Guys helped with JSON, so autoupdate should work.

**Comment by CharliePoole** [comment]
It's rather cool! I'm wondering about the naming, however.

On other platforms, when you install "nunit" you get the nunit-framework, not the console runner. So maybe "nunit-console" would be better.

As another option, Scoop is a bit like Chocolatey, since it's suitable for centrally installed executables. Maybe using the same names we use on chocolatey would make sense, as in

```
choco install nunit-console-runner
choco install nunit-console-with-extensions
```

I would lean toward the choco names if you are supporting both types of installs or just nunit-console if you want to have a single install (either with or without the extensions)

A further followup related to chocolatey is that you can install extensions with choco as well. We have to create speciffic packages to allow that, however.

**Comment by unickq** [comment]
@CharliePoole 
I didn't see NUnit console with extensions zip in the last release, that's why only one package here.

So you want me to rename `nunit` to `nunit-console`, am I right?

**Comment by CharliePoole** [comment]
I would suggest that, unless you want to deal with the more complicated issue of extensions being added separately. What does the @nunit/engine-team think?

The chocolatey and nuget packages we release are separate for console only and console plus extensions. They are both nupkg files but with different structures since chocolatey has slightly different rules from nuget. Since a nupkg is just a zip, you could open one of them and see what's there.

If you think separating the extensions is a good idea, then nunit-console is a good name for now. I used nunit-console-runner to suggest it's __only__ the runner and nunit-console-with-extensions to indicate that the "main" extensions are included.

If you wanted to allow something like `scoop install nunit-extension-teamcity` you would have to figure out where to store the extensions relative to the console installation in scoop, so that the nunit engine is able to find it. So that's a somewhat bigger task than just having one install.

Bottom line, IMO, nunit-console would be good for now, but wait for others to respond too.

**Comment by ChrisMaddock** [comment]
From memory, I think the zip does include the extensions. Can’t check on my phone, sorry.

I agree with the nunit-console naming idea. 😊

**Comment by mikkelbu** [comment]
I'm also for nunit-console.

Ps. The zip, i.e. https://github.com/nunit/nunit-console/releases/download/v3.9/NUnit.Console-3.9.0.zip, includes the extensions.

**Comment by unickq** [comment]
I guess we can close this issue now

**Comment by mikkelbu** [comment]
Ok. Closing and thanks for the work @unickq and for bringing Scoop to our attention 👍
